Sources/Widgets/Core/WidgetManager/index.js

```diff
@@ -253,30 +253,40 @@ function vtkWidgetManager(publicAPI, model) {
   // API public
   // --------------------------------------------------------------------------
 
-  async function updateSelection({ position }) {
-    model._selectionInProgress = true;
+  async function updateSelection(callData, fromTouchEvent) {
+    const { position } = callData;
     const { requestCount, selectedState, representation, widget } =
       await publicAPI.getSelectedDataForXY(position.x, position.y);
-    model._selectionInProgress = false;
 
     if (requestCount) {
       // Call activate only once
       return;
     }
 
+    function activateHandle(w) {
+      if (fromTouchEvent) {
+        // release any previous left button interaction
+        model._interactor.invokeLeftButtonRelease(callData);
```

Comment:
shouldn't it be conditional ? 
It seems dangerous to invoke a release event if no press event was previously called.
Some interactor observer may only have a callback to onLeftButonRelease.
Maybe that can be handled by the RWI by overloading the `invokeXXX` methods

Reply:
If `fromTouchEvent` is true, then there should have been a LeftButtonPress triggered just before. The way this works is that a touch event always triggers LeftButtonPress. Since the widget manager needs to do an (async) pick+selection, it can't abort the event chain, so the event may get picked up by an interactor style. Once the picking is finished and a widget is deemed to be selected, then we synthetically invoke LeftButtonRelease to release the focus on the interactor style so that we can grab focus on the now-selected widget.

Callback ordering for OnLeftButtonPress: `widgets -> widgetManager -> interactorStyle (if available)`.

This does have an issue where if picking takes too slow, then users may see out-of-order behavior. I'm open to suggestions on this.

Reply:
fair enough, the `if (fromTouchEvent)` block can behave as the `condition` I was looking for.